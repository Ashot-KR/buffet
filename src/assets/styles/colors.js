const colors = {
  darkBlue: '#1C5DE7',
  mediumBlue: '#005EEA',
  lightBlue: '#0097F6',
  blueGradient:
    'linear-gradient(to right top, #306ded, #2f78f1, #3283f3, #3a8ef6, #4598f7)',
  mediumGrey: '#979797',
  lightGrey: 'F6F6F6',
  lightGreyAlpha: 'rgba(0,0,0,0.1)',
  darkOrange: '#F64D0A',
  orangeBorder: 'rgba(255, 0, 0, 0.2)',
  orangeBkgd: 'rgba(255, 0, 0, 0.15)',
  white: '#ffffff',
};
export default colors;
