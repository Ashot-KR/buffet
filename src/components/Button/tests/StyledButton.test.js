import React from 'react';
/* eslint-disable */
import { mount } from 'enzyme';
import { enzymeFind } from 'styled-components/test-utils';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { StyledButton } from '../index';
import colors from '../../../assets/styles/Colors';

describe('<StyledButton />', () => {
  describe('<StyledButton /> attributes', () => {
    it('should render an <button> tag', () => {
      const wrapper = mount(<StyledButton />);
      const renderedComponent = enzymeFind(wrapper, StyledButton);
      expect(renderedComponent.type()).toEqual('button');
    });

    it('should have a className attribute', () => {
      const wrapper = mount(<StyledButton />);
      const renderedComponent = enzymeFind(wrapper, StyledButton);
      expect(renderedComponent.prop('className')).toBeDefined();
    });

    it('should adopt a valid attribute', () => {
      const id = 'test';
      const wrapper = mount(<StyledButton id={id} />);
      const renderedComponent = enzymeFind(wrapper, StyledButton);
      expect(renderedComponent.prop('id')).toEqual(id);
    });

    it('should not adopt an invalid attribute', () => {
      const wrapper = mount(<StyledButton attribute="test" />);
      const renderedComponent = enzymeFind(wrapper, StyledButton);
      expect(renderedComponent.prop('attribute')).toBeUndefined();
    });

    it('should adopt a default type button', () => {
      const wrapper = mount(<StyledButton />);
      const renderedComponent = enzymeFind(wrapper, StyledButton);
      expect(renderedComponent.prop('type')).toEqual('button');
    });

    it('should adopt a type submit', () => {
      const wrapper = mount(<StyledButton type="submit" />);
      const renderedComponent = enzymeFind(wrapper, StyledButton);
      expect(renderedComponent.prop('type')).toEqual('submit');
    });
  });

  describe('<StyledButton /> icons', () => {
    it('should render the add icon', () => {
      const tree = renderer.create(<StyledButton icon="add" />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render the add back', () => {
      const tree = renderer.create(<StyledButton icon="back" />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render the add trash', () => {
      const tree = renderer.create(<StyledButton icon="trash" />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render the primary style', () => {
      const tree = renderer.create(<StyledButton color="primary" />).toJSON();
      expect(tree).toHaveStyleRule('color', colors.white);
    });

    it('should render the secondary style', () => {
      const tree = renderer.create(<StyledButton color="secondary" />).toJSON();
      expect(tree).toHaveStyleRule('color', colors.darkBlue);
    });

    it('should render the cancel style', () => {
      const tree = renderer.create(<StyledButton color="cancel" />).toJSON();
      expect(tree).toHaveStyleRule('color', colors.mediumGrey);
    });

    it('should render the delete style', () => {
      const tree = renderer.create(<StyledButton color="delete" />).toJSON();
      expect(tree).toHaveStyleRule('color', colors.darkOrange);
    });
  });
});
