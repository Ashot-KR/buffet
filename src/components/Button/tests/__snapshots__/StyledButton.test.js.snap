// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledButton /> <StyledButton /> icons should render the add back 1`] = `
.c0 {
  height: 30px;
  padding: 0 30px;
  font-family: 'LatoSemiBold';
  font-size: 13px;
  line-height: 100%;
  border-radius: 3px;
  cursor: pointer;
  outline: 0;
  background-image: linear-gradient(to right top,#306ded,#2f78f1,#3283f3,#3a8ef6,#4598f7);
  color: #ffffff;
  border: 0;
}

.c0:hover {
  -webkit-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  -moz-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  box-shadow: inset 0 0 30px [object Object];
}

.c0:active {
  -webkit-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  -moz-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  box-shadow: inset 0 0 30px [object Object];
}

.c0::before {
  font-family: 'FontAwesome';
  font-weight: 600;
  font-size: 14px;
  margin-right: 10px;
  line-height: 100%;
  display: inline-block;
  vertical-align: middle;
}

.c0::before {
  content: '\\f053';
}

<button
  className="c0"
  color="primary"
  type="button"
/>
`;

exports[`<StyledButton /> <StyledButton /> icons should render the add icon 1`] = `
.c0 {
  height: 30px;
  padding: 0 30px;
  font-family: 'LatoSemiBold';
  font-size: 13px;
  line-height: 100%;
  border-radius: 3px;
  cursor: pointer;
  outline: 0;
  background-image: linear-gradient(to right top,#306ded,#2f78f1,#3283f3,#3a8ef6,#4598f7);
  color: #ffffff;
  border: 0;
}

.c0:hover {
  -webkit-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  -moz-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  box-shadow: inset 0 0 30px [object Object];
}

.c0:active {
  -webkit-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  -moz-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  box-shadow: inset 0 0 30px [object Object];
}

.c0::before {
  font-family: 'FontAwesome';
  font-weight: 600;
  font-size: 14px;
  margin-right: 10px;
  line-height: 100%;
  display: inline-block;
  vertical-align: middle;
}

.c0::before {
  content: '\\f067';
}

<button
  className="c0"
  color="primary"
  type="button"
/>
`;

exports[`<StyledButton /> <StyledButton /> icons should render the add trash 1`] = `
.c0 {
  height: 30px;
  padding: 0 30px;
  font-family: 'LatoSemiBold';
  font-size: 13px;
  line-height: 100%;
  border-radius: 3px;
  cursor: pointer;
  outline: 0;
  background-image: linear-gradient(to right top,#306ded,#2f78f1,#3283f3,#3a8ef6,#4598f7);
  color: #ffffff;
  border: 0;
}

.c0:hover {
  -webkit-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  -moz-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  box-shadow: inset 0 0 30px [object Object];
}

.c0:active {
  -webkit-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  -moz-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  box-shadow: inset 0 0 30px [object Object];
}

.c0::before {
  font-family: 'FontAwesome';
  font-weight: 600;
  font-size: 14px;
  margin-right: 10px;
  line-height: 100%;
  display: inline-block;
  vertical-align: middle;
}

.c0::before {
  content: '\\f014';
  vertical-align: inherit;
}

<button
  className="c0"
  color="primary"
  type="button"
/>
`;
